src/pages/blog/index.astro

```diff
@@ -6,16 +6,16 @@ import BlogCard from "../../sections/blog/BlogCard.astro";
 import { getLink } from "../../utils";
 import PageTitle from "../../components/PageTitle.astro";
 
-const posts = (await getCollection("blog")).sort(
-  (a, b) => a.data.pubDate.valueOf() - b.data.pubDate.valueOf()
-);
+const posts = (await getCollection("blog"))
+  .sort((a, b) => a.data.pubDate.valueOf() - b.data.pubDate.valueOf())
+  .filter((post) => !post.data.hidden);
```

Comment:
`!post.data.hidden` - negotion of the property; it means that all data with `hidden: false` or no `hidden` at all will be in the list.